### PR TITLE
Adding apigw-sfn-crud-terraform pattern

### PR DESCRIPTION
**Pull request by enmata**
*Issue #, if available:*
https://github.com/aws-samples/serverless-patterns/issues/2443

*Description of changes:*
Adding terraform version of apigw-sfn-crud pattern
https://serverlessland.com/patterns/apigw-sfn-crud

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: apigw-sfn-crud-terraform/main.tf [hunk 1]
```diff
@@ -0,0 +1,401 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 3.27"
```
Comment (bfreiberg) [inline apigw-sfn-crud-terraform/main.tf:5]: Please use the most recent version

File: apigw-sfn-crud-terraform/main.tf [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+}
+
+#################################################################
+# Step Function - State Machine
+#################################################################
+# Creating the Step Functions Machine
+resource "aws_sfn_state_machine" "API_SF_CRUD-tf_CF" {
+  name     = "API_SF_CRUD-tf_CF"
+  role_arn = aws_iam_role.API_SF_CRUD-tf_CFRole.arn
+  definition = file("./crud.asl.json")
```
Comment (bfreiberg) [inline apigw-sfn-crud-terraform/main.tf:290]: File is named `crud-asl.json`

**Review by bfreiberg — CHANGES_REQUESTED** [review]
I get an error when running `terraform apply´ due to CloudWatch Logs ARN not being specified. Please add instructions or link to documentation on how to do this.

╷
│ Error: error updating API Gateway Stage (ags-6omth9g63d-Prod): BadRequestException: CloudWatch Logs role ARN must be set in account settings to enable logging
│
│   with aws_api_gateway_stage.API_SF_CRUD-tf_API_Stage_Prod,
│   on main.tf line 222, in resource "aws_api_gateway_stage" "API_SF_CRUD-tf_API_Stage_Prod":
│  222: resource "aws_api_gateway_stage" "API_SF_CRUD-tf_API_Stage_Prod" {
│
╵

File: apigw-sfn-crud-terraform/README.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    ```
+    terraform apply
+    ```
+5. During the prompts:
+    * Enter yes
+6. Note the outputs from the deployment process. These contain the resource names and/or URLs which are used for testing.
+
+## How it works
+
+Amazon API Gateway creates a direct integration with AWS Step Functions utilizing a synchronous call. Step functions evaluates the path and method to choose the proper action. The action steps can be modified to meet your needs.
```
Comment (bfreiberg) [inline apigw-sfn-crud-terraform/README.md:40]:
```suggestion
Amazon API Gateway creates a direct integration with AWS Step Functions utilizing a synchronous call. The Step Functions state machine evaluates the path and method to choose the proper action. The action steps can be modified to meet your needs.
```

File: apigw-sfn-crud-terraform/README.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+## Testing
+
+*CRUD = Create, Read, Update, Delete*
+
+Once the application is deployed, use a tool like Postman or Curl to call the different CRUD endpoints. If you would like to import the Postman package, update the endpoint and import into Postman.
```
Comment (bfreiberg) [inline apigw-sfn-crud-terraform/README.md:46]: It isn't clear to me what is meant be updating the endpoint and importing it? Please also add an example of a `curl` request

File: apigw-sfn-crud-terraform/main.tf (hunk 1 above)
Comment (enmata) [reply apigw-sfn-crud-terraform/main.tf]: Updated, thanks.

File: apigw-sfn-crud-terraform/main.tf (hunk 2 above)
Comment (enmata) [reply apigw-sfn-crud-terraform/main.tf]: Reference now fixed, thanks

File: apigw-sfn-crud-terraform/README.md (hunk 3 above)
Comment (enmata) [reply apigw-sfn-crud-terraform/README.md]: More clear, change applied

File: apigw-sfn-crud-terraform/README.md (hunk 4 above)
Comment (enmata) [reply apigw-sfn-crud-terraform/README.md]: Added a rephrased version.

**Comment by enmata** [comment]
Changes applied. Please, could you check?

File: apigw-sfn-crud-terraform/main.tf [hunk 5]
```diff
@@ -0,0 +1,401 @@
...
+    type = "S"
+  }
+
+  tags = {
+    Name = "API_SF_CRUD-DB"
+  }
+}
+
+#################################################################
+# Step Function - State Machine
```
Comment (bfreiberg) [inline apigw-sfn-crud-terraform/main.tf:284]:
```suggestion
# Step Functions - State Machine
```

File: apigw-sfn-crud-terraform/main.tf [hunk 6]
```diff
@@ -0,0 +1,401 @@
...
+#################################################################
+# Creating the Step Functions Machine
```
Comment (bfreiberg) [inline apigw-sfn-crud-terraform/main.tf:286]:
```suggestion
# Creating the state machine
```

File: apigw-sfn-crud-terraform/example-pattern.json [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+  "cleanup": {
+    "text": [
+      "terraform destroy"
+    ]
+  },
+  "authors": [
+    {
+      "name": "Oriol Matavacas",
+      "image": "https://togithub.s3.eu-west-1.amazonaws.com/Oriol.jpg",
+      "bio": "Oriol Matavacas is a Sr. Solutions Architect at AWS based in Barcelona. Oriol primarily supporting customers on the journey to the Cloud. He enjoys building new solutions with scalability, availability and easy to maintain by using serverless.",
```
Comment (bfreiberg) [inline apigw-sfn-crud-terraform/example-pattern.json:57]:
```suggestion
      "bio": "Oriol Matavacas is a Senior Solutions Architect at AWS based in Barcelona. Oriol primarily supports customers on the journey to the Cloud. He enjoys building new solutions with scalability, availability and easy to maintain by using serverless.",
```

File: apigw-sfn-crud-terraform/example-pattern.json [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+    ]
+  },
+  "testing": {
+    "text": [
+      "See the GitHub repo for detailed testing instructions."
+    ]
+  },
+  "cleanup": {
+    "text": [
+      "terraform destroy"
```
Comment (bfreiberg) [inline apigw-sfn-crud-terraform/example-pattern.json:50]:
```suggestion
      "<code>terraform destroy</code>"
```

File: apigw-sfn-crud-terraform/example-pattern.json [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+    ]
+  },
+  "authors": [
+    {
+      "name": "Oriol Matavacas",
+      "image": "https://togithub.s3.eu-west-1.amazonaws.com/Oriol.jpg",
+      "bio": "Oriol Matavacas is a Sr. Solutions Architect at AWS based in Barcelona. Oriol primarily supporting customers on the journey to the Cloud. He enjoys building new solutions with scalability, availability and easy to maintain by using serverless.",
+      "linkedin": "https://www.linkedin.com/in/oriol-matavacas-rodriguez-b165868a",
```
Comment (bfreiberg) [inline apigw-sfn-crud-terraform/example-pattern.json:58]:
```suggestion
      "linkedin": "oriol-matavacas-rodriguez-b165868a",
```

File: apigw-sfn-crud-terraform/example-pattern.json [hunk 10]
```diff
@@ -0,0 +1,63 @@
+{
+  "title": "Amazon API Gateway, AWS Step Functions, to Amazon DynamoDB CRUD API",
+  "description": "Create a CRUD API with AWS Step Functions.",
+  "language": "Terraform",
```
Comment (bfreiberg) [inline apigw-sfn-crud-terraform/example-pattern.json:4]:
```suggestion
  "language": "",
```

File: apigw-sfn-crud-terraform/example-pattern.json [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+    "headline": "Create a CRUD API with AWS Step Functions.",
+    "text": [
+      "This stack creates a fully functioning CRUD API powered by Amazon API Gateway direct integration to AWS Step Functions and backed by Amazon DynamoDB."
+    ]
+  },
+  "gitHub": {
+    "template": {
+      "repoURL": "https://github.com/aws-samples/serverless-patterns/tree/main/apigw-sfn-crud-terraform",
+      "templateURL": "serverless-patterns/apigw-sfn-crud-terraform",
+      "projectFolder": "apigw-sfn-crud-terraform,
```
Comment (bfreiberg) [inline apigw-sfn-crud-terraform/example-pattern.json:17]:
```suggestion
      "projectFolder": "apigw-sfn-crud-terraform",
```

File: apigw-sfn-crud-terraform/crud-asl.json [hunk 12]
```diff
@@ -0,0 +1,105 @@
+{
+  "Comment" : "A description of my state machine",
```
Comment (bfreiberg) [inline apigw-sfn-crud-terraform/crud-asl.json:2]: Please use a more descriptive comment

File: apigw-sfn-crud-terraform/README.md [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+    * Enter yes
+6. Note the outputs from the deployment process. These contain the resource names and/or URLs which are used for testing.
+
+## How it works
+
+Amazon API Gateway creates a direct integration with AWS Step Functions utilizing a synchronous call. The Step Functions state machine evaluates the path and method to choose the proper action. The action steps can be modified to meet your needs.
+
+## Testing
+
+*CRUD = Create, Read, Update, Delete*
```
Comment (bfreiberg) [inline apigw-sfn-crud-terraform/README.md:44]: Move this up to where CRUD is first mentioned to explain the acronym there

File: apigw-sfn-crud-terraform/main.tf (hunk 5 above)
Comment (enmata) [reply apigw-sfn-crud-terraform/main.tf]: Fixed, thanks

File: apigw-sfn-crud-terraform/example-pattern.json (hunk 7 above)
Comment (enmata) [reply apigw-sfn-crud-terraform/example-pattern.json]: Typos fixed

File: apigw-sfn-crud-terraform/example-pattern.json (hunk 8 above)
Comment (enmata) [reply apigw-sfn-crud-terraform/example-pattern.json]: Change applied

File: apigw-sfn-crud-terraform/example-pattern.json (hunk 9 above)
Comment (enmata) [reply apigw-sfn-crud-terraform/example-pattern.json]: Corrected

File: apigw-sfn-crud-terraform/example-pattern.json (hunk 10 above)
Comment (enmata) [reply apigw-sfn-crud-terraform/example-pattern.json]: OK, done

File: apigw-sfn-crud-terraform/example-pattern.json (hunk 11 above)
Comment (enmata) [reply apigw-sfn-crud-terraform/example-pattern.json]: typo fixed

File: apigw-sfn-crud-terraform/crud-asl.json (hunk 12 above)
Comment (enmata) [reply apigw-sfn-crud-terraform/crud-asl.json]: Description rephrased

**Comment by enmata** [comment]
Changes applied, Please, do you mind to check again?

**Pull request merged**